pkg/calculator/handler.go

```diff
@@ -43,14 +46,30 @@ func NewEmissionCalculator(ctx context.Context, eventbus bus.Bus) *EmissionCalcu
 		logger.Error("unable to get v2 Emission Factors, falling back to v1", "error", err)
 	}
 
-	return &EmissionCalculator{
-		eventbus,
-		logger,
+	return &CalculatorHandler{
+		Bus:    b,
+		logger: logger,
+	}
+}
+
+// Stop is used to fulfill the EventHandler interface and all clean up
+// functionality should be run in here
+func (c *CalculatorHandler) Stop(ctx context.Context) {}
+
+// Handle is used to fulfill the EventHandler interface and recives an event
+// when handler is subscribed to it. Currently only handles v1.MetricsCollectedEvent
+func (c *CalculatorHandler) Handle(ctx context.Context, e *bus.Event) {
+	switch e.Type {
```

Comment:
Can we just return the MetricsCollectedEvent?

Reply:
this is just to manage that we dont by mistake call the handle with the wrong event

Reply:
tbh that feels a bit over engineering to me

Reply:
the other way of doing this is:
```golang
if e.Type == v1.MetricsCollectedEvent {
    c.handleEvent(e)
}
```

But I thought that as golang preffered type checking with switch cases (from my understanding) it made more sense to do it this way. I can change it to the above if you would like

Reply:
Agreed, let's keep it!